### ISSUE TITLE
cmd.* cwd does not escape spaces. 2017.7.2

### ISSUE THREAD

**Issue by whytewolf** [saltstack/salt#44315]
### Description of Issue/Question
This was first noticed in windows, while trying to cwd to `c:\Program Files`, however, I have also found it to be the case in Linux.
a cmd.script or cmd.run with a cwd that has a space in it will error out. Such as 
`cmd.script '/srv/test 2/test.sh' cwd='/srv/test 2'` if you remove the cwd with the space the command will run fine. 

```
root@master test 2]# salt-call cmd.script '/srv/test 2/test.sh'
local:
    ----------
    pid:
        27714
    retcode:
        0
    stderr:
    stdout:
        test
[root@master test 2]# salt-call cmd.script '/srv/test 2/test.sh' cwd='/srv/test 2'
local:
    ----------
    pid:
        27796
    retcode:
        127
    stderr:
        /bin/bash: /srv/test: No such file or directory
    stdout:
[root@master test 2]# salt-call cmd.script '/srv/test.sh'
local:
    ----------
    pid:
        27940
    retcode:
        0
    stderr:
    stdout:
        test
[root@master test 2]# salt-call cmd.script '/srv/test.sh' cwd='/srv/test 2'
local:
    ----------
    pid:
        28022
    retcode:
        127
    stderr:
        /bin/bash: /srv/test: No such file or directory
    stdout:
```

### Setup
default setup can be done masterless. and without states. although the error does carry over into the cmd.* states as well. 

### Steps to Reproduce Issue
create a test script to run I typically use a file containing `echo 'test'`
put that file in two different directories. one with a space one without. such as `/srv/test` and `/srv/test 2`
run each cmd with and without `cwd='/srv/test 2'`

### Versions Report
```
salt-call --versions
Salt Version:
           Salt: 2017.7.2

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.5 (default, Aug  4 2017, 00:39:18)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.4.1708 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-693.2.1.el7.x86_64
         system: Linux
        version: CentOS Linux 7.4.1708 Core
```

**Comment by whytewolf** [comment]
ZD-1923

**Comment by Ch3LL** [comment]
Looks like i'm able to replicate this behavior and doesn't help to escape the space either.

**Comment by rallytime** [comment]
@whytewolf This should be fixed with #45134.
